.github/workflows/ci.yml

```diff
@@ -98,50 +120,92 @@ jobs:
+
+  test_py:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [2.7, 3.7]
+
+    # Skip redundant checks for binary releases
+    if: "!startsWith(github.ref, 'refs/heads/release/')"
+
+    name: Test Python ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - uses: actions-rs/toolchain@v1
+        id: toolchain
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
 
-      - name: Deploy
-        if: github.ref == 'refs/heads/master'
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Dependencies
+        run: pip install -U pytest
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: pytest-${{ runner.os }}-${{ steps.toolchain.outputs.rustc_hash }}-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Build and Install Library
+        run: pip install -v --editable py
         env:
-          GITHUB_TOKEN: ${{ secrets.DATA_SCHEMAS_GITHUB_TOKEN }}
-        run: |
-          git config --global user.name "$(git log -1 --pretty=format:%an $GITHUB_SHA)"
-          git config --global user.email "$(git log -1 --pretty=format:%ae $GITHUB_SHA)"
-          git clone https://getsentry-bot:$GITHUB_TOKEN@github.com/getsentry/sentry-data-schemas
-          cd sentry-data-schemas/
-          mkdir -p ./relay/
-          mv ../event.schema.json relay/event.schema.json
-          git add relay/event.schema.json
-
-          echo "attempting commit"
-          if ! git commit -m "getsentry/relay@$GITHUB_SHA" ; then
-            echo "Stopping, no changes"
-            exit 0
-          fi
-
-          for i in 1 2 3 4 5; do
-            echo "git push; Attempt $i"
-            if git push; then
-              exit 0
-            fi
-
-            git pull --rebase
-          done
-
-          echo "Failed to push"
-          exit 1
-
-  metrics_docs:
-    name: Metrics Docs
+          RELAY_DEBUG: 1
+
+      - name: Run Python Tests
+        run: pytest -v py
+
+  test_integration:
+    name: Integration Tests
     runs-on: ubuntu-latest
+
+    # Skip redundant checks for library releases
+    if: "!startsWith(github.ref, 'refs/heads/release-library/')"
+
+    services:
+      redis: # https://docs.github.com/en/actions/guides/creating-redis-service-containers
+        image: redis
+        ports:
+          - 6379:6379
+
+      zookeeper:
+        image: confluentinc/cp-zookeeper
+        env:
+          ZOOKEEPER_CLIENT_PORT: 2181
+
+      kafka:
+        image: confluentinc/cp-kafka
+        env:
+          KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+          KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://localhost:9092
+          KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
```

Comment:
You may want to apply the changes from https://github.com/getsentry/sentry/pull/20780 here

Reply:
Thanks, that's a good hint! I'll add `KAFKA_OFFSETS_TOPIC_NUM_PARTITIONS`. We can skip setting the log retention as the service and its volumes never live longer than a few minutes.

Reply:
I'm actually realizing that our test setup manually ensures a single partition already in application code since I didn't know about that setting. Still adding this as another layer of protection 👍